### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by jiangk**
Added python future library, various fixes for py2/py3 compatibility.

For review, it'll be easier if you look at the individual commit diffs instead of everything at once.

Tested macos with py2/3. Still need to manually test windows/linux. Fixes for those will come in a later commit.

### CODE REVIEW

File: pyu2f/convenience/customauthenticator.py [hunk 1]
```diff
@@ -235,5 +235,9 @@ def _CallPlugin(self, cmd, input_json):
   def _SHA256(self, string):
     """Helper method to perform SHA256."""
     md = hashlib.sha256()
-    md.update(string)
+    md.update(string.encode())
     return md.digest()
+
+  def _base64encode(self, bytes_data):
```
Comment (gdasher) [inline pyu2f/convenience/customauthenticator.py]: For consistency, lets call this _Base64Encode

File: pyu2f/hidtransport.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def ToWireFormat(self):
       ret[4] = self.cmd
       struct.pack_into('>H', ret, 5, self.size)
       ret[7:7 + len(self.payload)] = self.payload
-      return map(int, ret)
+      return list(map(int, ret))
```
Comment (gdasher) [inline pyu2f/hidtransport.py]: just curious, why is this necessary in python3?

Comment (jiangk) [reply pyu2f/hidtransport.py]: It's because map() returns a map object in python3, but it returns a list in python2.

**Comment by jiangk** [comment]
Bumping for review of windows changes, thanks!
